.github/workflows/docker.yml

```diff
@@ -23,40 +23,36 @@ jobs:
         run: |
           docker-compose --version
 
-      - name: Build docker
-        run: docker-compose build
+      - name: Build base image, start containers, and wait for services
+        run: |
+          docker-compose up --build -d
```

Comment:
It's been so long now that I can't remember how this stuff works. 
*Before*, it used just build but not start all the services. Now, with this command, it's basically the exact equivalent of doing `docker-compose build && docker-compose up -d`. 

So now that the daemon is started, you can use `docker-compose exec` instead of `docker-compose run` because `exec` expects and assume the daemon to be up and running. But if it isn't it just exits non-zero.
```sh
▶ docker-compose build --parallel
...

▶ docker-compose exec web make localecompile

▶ echo $?
1
```

Whereas `docker-compose run` is a lot more convenient. Not only does it take care of the NOT having to rely on `urlwait`, it only starts the services (aka. `depends_on`) that it needs:

```
▶ docker-compose stop

▶ docker-compose build --parallel

▶ docker-compose run web make localecompile
Starting kuma_mockga_1 ... done
Starting kuma_mysql_1  ... done
Starting kuma_redis_1  ... done
Starting kuma_minio_1  ... done
Creating kuma_web_run  ... done
...

▶ echo $?
0
```

So `docker-compose run` feels like a better tool. It does what `exec` does but it takes care of not having to wait for a background daemon to maybe have been started. 

Granted, even `docker-compose run` isn't protected from *not* having to use `urlwait`. 
When you type `docker-compose run` it will START `mysql` (and `worker` and `minio` etc) before it proceeds with the rest of the command (e.g. `make localecompile`). But just because it started the service doesn't mean the service has fully started as us humans know it. So if `localcompile` depends on talking to MySQL could easily crash with `docker-compose build && docker-compose run web make localecompile` if that was the case. 

There might be a small overhead of `docker-compose run` because it's so much more careful than `exec` but I think I prefer `run` and to *not* run `docker-compose up -d`. 

Does that make sense? I.e. drop the MySQL `urlwait` and go back to the old way but without good idea of adding `-T` to the `docker-compose run ...` commands.

Reply:
No, `docker-compose run` will leave them running. I tested this. I think the `localecompile` would have getting MySQL started. That's how we survived before. But, I can see the possible problem still exists even with `run`. The solution is that you have to do this:

```diff
+ docker-compose run -T web urlwait mysql://root:kuma@mysql:3306/developer_mozilla_org 30
  docker-compose run -T web ./manage.py migrate
```
to be entirely safe. 
Then the first command will kill two birds with one stone, in favor of the second command.

Reply:
@peterbe 👍 I'll switch back to the original `.github/workflows/docker.yml` and add the `urlwait` as you suggested.

Reply:
Now I get why the `master` branch isn't having the same problems with MySQL not yet being fully awake. Before, well, in `master` as we speak, it would first do the `buildstatic` which takes 20 minutes so that gives MySQL plenty of time to start. 

I approve of your `docker.yml` now.